packages/@aws-cdk/core/lib/construct-compat.ts

```diff
@@ -416,7 +417,7 @@ export class ConstructNode {
    * An immutable array of metadata objects associated with this construct.
    * This can be used, for example, to implement support for deprecation notices, source mapping, etc.
    */
-  public get metadata() { return this._actualNode.metadata as cxapi.MetadataEntry[]; }
+  public get metadata() { return this._actualNode.metadata as cxschema.MetadataEntry[]; }
```

Comment:
This is a breaking change on a stable part of the CDK. This might be ok as the number of CDK users using this API is likely to be very low, and the cost of fixing this on the user's end is very simple.